.azure/app-cloud-e2e.yml

```diff
@@ -0,0 +1,144 @@
+# Python package
+# Create and test a Python package on multiple Python versions.
+# Add steps that analyze code, save the dist with the build record, publish to a PyPI-compatible index, and more:
+# https://docs.microsoft.com/azure/devops/pipelines/languages/python
+
+trigger:
+  tags:
+    include:
+      - '*'
+  branches:
+    include:
+      - "master"
+      - "release/*"
+      - "refs/tags/*"
+
+pr:
+  - "master"
+  - "release/*"
+
+variables:
+  # variables are automatically exported as environment variables so this will override pip's default cache dir
+  - name: pip_cache_dir
+    value: $(Pipeline.Workspace)/.pip
+
+jobs:
+  - job: App_cloud_e2e_testing
+    pool:
+      vmImage: 'ubuntu-latest'
+    timeoutInMinutes: "30"
+    cancelTimeoutInMinutes: "2"
+    strategy:
+      matrix:
+        'App: v0_app':
+          name: "v0_app"
+        'App: boring_app':
+          name: "boring_app"
+        'App: template_streamlit_ui':
+          name: "template_streamlit_ui"
+#        'App: template_react_ui':  # TODO: clarify visibility private/public
+#          name: "template_react_ui"
+        'App: template_jupyterlab':  # TODO: clarify where these files lives
+          name: "template_jupyterlab"
+        'App: idle_timeout':
+          name: "idle_timeout"
+        'App: collect_failures':
+          name: "collect_failures"
+        'App: custom_work_dependencies':
+          name: "custom_work_dependencies"
+        'App: drive':
+          name: "drive"
+        'App: payload':
+          name: "payload"
+        'App: commands_and_api':
+          name: "commands_and_api"
+    workspace:
+      clean: all
+    steps:
+    - task: UsePythonVersion@0
+      inputs:
+        versi`onSpec: '3.8'
+
+    - script: |
+        ID=$(date +%s)
+        echo "##vso[task.setvariable variable=id;isOutput=true]$ID"
+      # TODO: parse the PR number
+      name: local
+
+    - task: Cache@2
+      inputs:
+        key: 'pip | "$(name)" | requirements/app/base.txt'
+        restoreKeys: |
+          pip | "$(Agent.OS)"
+        path: $(pip_cache_dir)
+      displayName: Cache pip
+
+    - bash: |
+        python --version
+        pip --version
+      displayName: 'Info'
+
+    - bash: python -m pip install -r requirements/app/devel.txt --quiet --find-links ${TORCH_URL}
+      env:
+        TORCH_URL: https://download.pytorch.org/whl/cpu/torch_stable.html
+      displayName: 'Install dependencies'
+
+    - bash: |
+        python -m pip install playwright
+        python -m playwright install --with-deps
+      displayName: 'Install Playwright system dependencies'
+
+    - bash: pip install -e .
```

Comment:
It shouldn't be installing from current directory, as the PL or APP can individually advance form the what `.` points to in the meta package. The latest e2e failure that blocked our release was due to this. We should build the package the same we are building it for releasing the package. Perhaps we can do it as another PR ?

Reply:
Let's do it with another PR =)